notebook/converted/bad_lr_performance.py

```diff
@@ -0,0 +1,382 @@
+plot += ggtitle('Sklearn LR confusion matrix')
+plot
+
+
+# ## Prediction distributions
+
+# In[92]:
+
+
+deep_net_predicted = collections.Counter(model_predictions['deep_net']['predicted'])
+deep_net_true = predicted = collections.Counter(model_predictions['deep_net']['true'])
+print(deep_net_predicted, deep_net_true)
+
+
+# In[93]:
+
+
+three_layer_net_predicted = collections.Counter(model_predictions['three_layer_net']['predicted'])
+three_layer_net_true = predicted = collections.Counter(model_predictions['three_layer_net']['true'])
+print(three_layer_net_predicted, three_layer_net_true)
+
+
+# In[94]:
+
+
+pytorch_lr_predicted = collections.Counter(model_predictions['pytorch_lr']['predicted'])
+pytorch_lr_true = predicted = collections.Counter(model_predictions['pytorch_lr']['true'])
+print(pytorch_lr_predicted, pytorch_lr_true)
+
+
+# In[95]:
+
+
+skl_lr_predicted = collections.Counter(model_predictions['logistic_regression']['predicted'])
+skl_lr_true = predicted = collections.Counter(model_predictions['logistic_regression']['true'])
+print(skl_lr_predicted, skl_lr_true)
+
+
+# In[101]:
+
+
+pred_df = pd.DataFrame([deep_net_predicted, three_layer_net_predicted, pytorch_lr_predicted, skl_lr_predicted])
+pred_df['model'] = ['five_layer_net', 'three_layer_net', 'pytorch_lr', 'skl_lr']
+pred_df['pred_or_true'] = 'pred'
+pred_df
+
+
+# In[102]:
+
+
+true_df = pd.DataFrame([deep_net_true, three_layer_net_true, pytorch_lr_true, skl_lr_true])
+true_df['model'] = ['five_layer_net', 'three_layer_net', 'pytorch_lr', 'skl_lr']
+true_df['pred_or_true'] = 'true'
+true_df
+
+
+# In[106]:
+
+
+all_df = pd.concat([pred_dict, true_dict])
+all_df
+
+
+# In[120]:
+
+
+plot = ggplot(all_df, aes(x='factor(pred_or_true)', y='Blood', fill='model'))
+plot += geom_bar(stat='identity')
+plot += facet_grid('model ~ .')
+print(plot)
+
+
+# In[121]:
+
+
+plot = ggplot(all_df, aes(x='factor(pred_or_true)', y='Breast', fill='model'))
+plot += geom_bar(stat='identity')
+plot += facet_grid('model ~ .')
+print(plot)
+
+
+# In[122]:
+
+
+plot = ggplot(all_df, aes(x='factor(pred_or_true)', y='Stem Cell', fill='model'))
+plot += geom_bar(stat='identity')
+plot += facet_grid('model ~ .')
+print(plot)
+
+
+# In[123]:
+
+
+plot = ggplot(all_df, aes(x='factor(pred_or_true)', y='Cervix', fill='model'))
+plot += geom_bar(stat='identity')
+plot += facet_grid('model ~ .')
+print(plot)
+
+
+# In[124]:
+
+
+plot = ggplot(all_df, aes(x='factor(pred_or_true)', y='Brain', fill='model'))
+plot += geom_bar(stat='identity')
+plot += facet_grid('model ~ .')
+print(plot)
+
+
+# ## Conclusion
+# There doesn't seem to be anything significantly different between the prediction distributions of the deep models and the linear models. The linear ones are just wrong more regardless of class.
+# 
+# Actually the differences between the five layer net and three layer net in terms of their prediction strategy are interesting (they perform better with respect to different classes).
```

Comment:
Maybe if you used them to build an ensemble you could do even better! 😂 

(I know maximizing performance isn't really your goal here, mostly just joking - but it is interesting that they perform so differently)

Reply:
Yeah probably! I'm surprised too, I'd expect the same model class to have the same strategy for optimizing the loss, but apparently not